mongo/integration/index_view_test.go

```diff
@@ -607,6 +607,37 @@ func TestIndexView(t *testing.T) {
 		}
 		assert.Nil(mt, cursor.Err(), "cursor error: %v", cursor.Err())
 	})
+	mt.Run("dropKey one", func(mt *mtest.T) {
```

Comment:
This test would be easier to extend if we parameterized the testing conditions. For example:

```go
tests := []struct {
	name   string
	models []mongo.IndexModel
	index  any
}{
	{
		name: "custom index name",
		models: []mongo.IndexModel{
			{
				Keys:    bson.Raw(bsoncore.NewDocumentBuilder().AppendInt32("username", 1).Build()),
				Options: options.Index().SetUnique(true).SetName("myidx"),
			},
		},
		index: bson.D{{"username", 1}},
	},
}

for _, test := range tests {
	t.Run(test.name, func(t *testing.T) {
		iv := mt.Coll.Indexes()

		indexNames, err := iv.CreateMany(context.Background(), test.models)

		assert.NoError(t, err)
		assert.Equal(mt, 2, len(indexNames), "expected 2 index names, got %v", len(indexNames))

		_, err = iv.DropKeyOne(context.Background(), test.index)
		assert.Nil(mt, err, "DropOne error: %v", err)

		cursor, err := iv.List(context.Background())
		assert.Nil(mt, err, "List error: %v", err)
		for cursor.Next(context.Background()) {
			var idx index
			err = cursor.Decode(&idx)
			assert.Nil(mt, err, "Decode error: %v (document %v)", err, cursor.Current)
			assert.NotEqual(mt, indexNames[1], idx.Name, "found index %v after dropping", indexNames[1])
		}
		assert.Nil(mt, cursor.Err(), "cursor error: %v", cursor.Err())
	})
}
```

Additionally, we should add more tests to the matrix other than "custom index name", such as "generated index name".